lib/ansible/playbook/helpers.py

```diff
@@ -112,10 +112,11 @@ def load_list_of_tasks(ds, play, block=None, role=None, task_include=None, use_h
                 # 1. the user set the 'static' option to true
                 # 2. one of the appropriate config options was set
                 # 3. the included file name contains no variables, and has no loop
-                is_static = t.static or \
-                            C.DEFAULT_TASK_INCLUDES_STATIC or \
-                            (use_handlers and C.DEFAULT_HANDLER_INCLUDES_STATIC) or \
-                            not templar._contains_vars(t.args.get('_raw_params')) and t.loop is None
+                if t.static is not None:
+                    is_static = t.static
```

Comment:
What about the case where the include filename is based on a var? If there is a var in the name, we either need to check to see if the var is available now (which might still cause problems for plays where the var is changed later before the include happens), raise an error if is_static == True when there's a var in the name, or fall back to is_static = False when there's a var.